pkg/api/steve/supportconfigs/generator.go

```diff
@@ -66,22 +84,14 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 		// if usePAYG query parameter exist that means we are using the new Pay-As-You-Go offering CSP billing adapter
 		cspChartNamespace = cspadapter.PAYGChartNamespace
 		cspChartName = cspadapter.PAYGChartName
+		h.checkAuthorization(cspChartNamespace, cspMeteringArchiveConfigmap, writer, request)
```

Comment:
so you check the authorization for the archive map here, then the general config after the else block... what happens if this check fails? Will it do anything other than log out the permissions error? by moving this code block to its own function (which makes sense given that its called twice), have  you changed the behavior when the check fails since the return will just pop out of the new function and not the ServeHttp function? though the ReturnHTTPError gets called regardless...

Reply:
This is retaining the existing behavior, that if user have no permission to access the necessary confirmaps needed to create the supportconfig bundle, the code will fail fast. For PAYG supportconfig, both maps are needed, whereas for BOYL, only the config one is required.